tests/weights/test_dweights_precalibration_checkpoints.py

```diff
@@ -0,0 +1,129 @@
+    expected_df_pass = spark_session.createDataFrame(
+        data=[
+            # fmt: off
+            #    id     country         num_households
+            #                                   groupby     sample_group
+            #                                                           design_weight_1
+            #                                                                       design_weight_2
+            #                                                                                   not_positive_or_null
+                (1,     'england',      3,      3.0,        'new',      7.0,        3.0,        None),
+                (1,     'england',      3,      3.0,        'new',      7.0,        3.0,        None),
+                (1,     'england',      3,      3.0,        'new',      7.0,        3.0,        None),
+
+                (1,     'england',      3,      3.0,        'old',      4.0,        5.0,        None),
+                (1,     'england',      3,      3.0,        'old',      4.0,        5.0,        None),
+                (1,     'england',      3,      3.0,        'old',      4.0,        5.0,        None),
+            # fmt: on
+        ],
+        schema=schema,
+    )
+    for column in ["design_weight_1", "design_weight_2"]:
+        expected_df_pass = expected_df_pass.withColumn(column, F.col(column).cast(DecimalType(38, 20)))
+    input_df_pass = expected_df_pass.drop("not_positive_or_null")
+
+    # with pytest.raises(DesignWeightError):
+    #     validate_design_weights(
+    #         df=input_df_pass,
+    #         num_households_by_cis_column="num_households",
+    #         num_households_by_country_column="num_households",
+    #         swab_weight_column="design_weight_1",
+    #         antibody_weight_column="design_weight_2",
+    #         cis_area_column="country",
+    #         country_column="country",
+    #         swab_group_by_columns=["country", "groupby"],
+    #         antibody_group_by_columns=["country", "groupby"],
+    #         rounding_value=18,
+    #     )
+
+    #
+
+    (
+        swab_weight_column_type,
+        antibody_weight_column_type,
+        antibody_design_weights_sum_to_population,
+        swab_design_weights_sum_to_population,
+        check_negative_design_weights,
+        check_null_design_weights,
+        swab_design_weights_inconsistent_within_group,
+        antibody_design_weights_inconsistent_within_group,
+    ) = validate_design_weights(
+        df=input_df_pass,
+        num_households_by_cis_column="num_households",
+        num_households_by_country_column="num_households",
+        swab_weight_column="design_weight_1",
+        antibody_weight_column="design_weight_2",
+        cis_area_column="country",
+        country_column="country",
+        swab_group_by_columns=["country", "groupby"],
+        antibody_group_by_columns=["country", "groupby"],
+        rounding_value=18,
+    )
+
+    import pdb
+
+    pdb.set_trace()
```

Comment:
Debugging statements and lots of commented out code here - please delete these if no longer necessary